cpp/src/arrow/dataset/scanner_test.cc

```diff
@@ -128,6 +128,18 @@ class TestScanner : public DatasetFixtureMixinWithParam<TestScannerParams> {
     AssertScanBatchesEquals(expected.get(), scanner.get());
   }
 
+  void AssertScanForAugmentedFields(std::shared_ptr<Scanner> scanner) {
+    auto result = scanner.get()->ToTable();
+    if (result.ok()) {
```

Comment:
Shouldn't `result` be OK here?

Reply:
I am not very sure about what you are asking. Here, we are just checking whether the `ToTable()` method returns `Result::OK` by calling the `ok()` method so that we can proceed with the test.

Reply:
Using ASSERT_OK_AND_ASSIGN here now.

Reply:
Yeah, ASSERT_OK_AND_ASSIGN is what I was expecting, thanks